From 41ab262d6148cc7adda524569680a755e57c211c Mon Sep 17 00:00:00 2001
From: Henrik Riomar <henrik.riomar@gmail.com>
Date: Thu, 6 Oct 2022 15:43:32 +0200
Subject: [PATCH] KeyType.EC_EDWARDS in test_sign_eddsa

The KeyType should be EC_EDWARDS for Ed25519

Fixes:
 BotanECDSA.cpp(301): ECDSA key generation failed

i.e. SoftHSM goes into BotanECDSA.cpp instead of BotanEDDSA.cpp where
1.3.101.112 is handled
---
 tests/test_ecc.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/test_ecc.py b/tests/test_ecc.py
index 30f685b..1c39dfd 100644
--- a/tests/test_ecc.py
+++ b/tests/test_ecc.py
@@ -149,7 +149,7 @@ class ECCTests(TestCase):
 
     @requires(Mechanism.EC_EDWARDS_KEY_PAIR_GEN, Mechanism.EDDSA)
     def test_sign_eddsa(self):
-        parameters = self.session.create_domain_parameters(KeyType.EC, {
+        parameters = self.session.create_domain_parameters(KeyType.EC_EDWARDS, {
             # use "Ed25519" once https://github.com/wbond/asn1crypto/pull/134
             # is merged
             Attribute.EC_PARAMS: encode_named_curve_parameters('1.3.101.112')
-- 
2.38.0

